File: Spike_Config_FINALE/Extra/EXAMPLE_create_txt_ramp.m
%%  Simple script to create a matrix with 2 columns to be plotted in Spike2 as a trace
%%%%%
% Author: Paolo Tecchio

clear
clc

% Col 1 = x (time points), Col 2 = y (whatever)

%must be columns
%x points or time points
matrix(:,1) = linspace(0,40,1000);

%y points
matrix(:,2) = sin(matrix(:,1))*20; %20 is the amplitude
matrix = round(matrix,2);
plot(matrix(:,1),matrix(:,2));

%save the txt
[fname,pth] = uiputfile('*.txt'); % Type in name of the file.
writematrix(matrix,[pth fname]);

File: Spike_Config_FINALE/Extra/EXAMPLE_create_txt_ramp.py
#   Simple script to create a matrix with 2 columns to be plotted in Spike2 as a trace
#   Author: Paolo Tecchio

import numpy as np
import matplotlib.pyplot as plt
from tkinter import filedialog

# Create a matrix with 2 columns
# Column 1 = x (time points), Column 2 = y (whatever)
matrix = np.zeros((1000, 2))

# x points or time points
matrix[:, 0] = np.linspace(0, 40, 1000)

# y points
matrix[:, 1] = np.sin(matrix[:, 0]) * 20  # 20 is the amplitude
matrix = np.round(matrix, 2)

# Save the data to a text file
file_path = filedialog.asksaveasfilename(defaultextension=".txt", filetypes=[("Text Files", "*.txt")])

if file_path:
    np.savetxt(file_path, matrix, delimiter=',', comments='')

# Plot the data
plt.plot(matrix[:, 0], matrix[:, 1])
plt.xlabel('Time Points')
plt.ylabel('Y Values')
plt.show()

